pyvisa-py/tcpip.py

```diff
@@ -381,21 +380,22 @@ def read(self, count):
             return (out + parts[0] + end_byte,
                     constants.StatusCode.success_termination_character_read)
 
-        # initial 'select_timout' is same as timeout or max 2s, so when no data arrived then max block time
-        select_timout = min(timeout, 2.0)
-        while now - start <= timeout:
-            # use select to wait for read ready, max `select_timout` seconds
-            r, w, x = select.select([self.interface], [], [], select_timout)
+        # initial 'select_timout' is half of timeout or max 2 secs, so when no data arrived then this is max block time
+        select_timout = min(timeout/2.0, 2.0)
+        # time, when loop shall finish
+        finish_time = time.time() + timeout
+        while time.time() <= finish_time:
+            # use select to wait for read ready, max `select_timout` seconds, min is 50ms
+            r, w, x = select.select([self.interface], [], [], max(select_timout, 0.05))
 
             last = b''
             if self.interface in r:
                 last = read_fun(chunk_length)
 
             if not last:
                 # can't read chunk or timeout
-                # `select_timout` decreased to 0.01 sec
-                select_timout = 0.01
-                now = time.time()
+                # `select_timout` decreased to 50% of previous, min wait time is 100ms
```

Comment:
min is 50 ms, no ?

Reply:
yes, I tested 100ms and 50ms for 'timeout delay', so it is timeouted in interval '<timeout, timeout + 50ms)', typicaly <40ms delay.